src/DotNetCore.CAP.MongoDB/IStorage.MongoDB.cs

```diff
@@ -59,7 +59,47 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
                     cancellationToken: cancellationToken);
             }
 
-            _logger.LogDebug("Ensuring all create database tables script are applied.");
+            var received_message_index_names = new string[] {
+                nameof(ReceivedMessage.Name), nameof(ReceivedMessage.Added), nameof(ReceivedMessage.ExpiresAt),
+                nameof(ReceivedMessage.StatusName), nameof(ReceivedMessage.Retries), nameof(ReceivedMessage.Version) };
+
+            var published_message_index_names = new string[] {
+                nameof(PublishedMessage.Name), nameof(PublishedMessage.Added), nameof(PublishedMessage.ExpiresAt),
+                nameof(PublishedMessage.StatusName), nameof(PublishedMessage.Retries), nameof(PublishedMessage.Version) };
+
+            await Task.WhenAll(
+                TryCreateIndexes<ReceivedMessage>(_options.ReceivedCollection, received_message_index_names),
+                TryCreateIndexes<PublishedMessage>(_options.PublishedCollection, published_message_index_names)
+                );
+
+            _logger.LogDebug("Ensuring all create database tables script are applied.");
+
+            async Task TryCreateIndexes<T>(string col_name, string[] index_names)

```

Comment:
Please using Lower Camel Case to replace Under Score Case for naming convention.

`async Task TryCreateIndexes<T>(string collectionName, string[] indexNames)`